### ISSUE TITLE
Time of music

### ISSUE THREAD

**Issue by Sodayeyeye** [AMAAI-Lab/mustango#14]
There is a question about the time to generate music, can only generate 10s at most?

**Comment by dorienh** [comment]
Until Mustango 2.0 comes out that is the limit indeed.

**Comment by kkaiwwana** [comment]
hi @Sodayeyeye, to generate longer music under current setup is actually pretty doable. Tranfering a super simple "copy & paste" trick that I introduced in this [paper](https://arxiv.org/abs/2407.19468) (wasn't for audio gen tho) can generate arbitrary length audio in this case. And more importantly, it's training-free! I implemented it for mustango to generate arbitrary long music, which has a varying style while remains favorable temporal consistency. The implementation can be found in this forked [repo](https://github.com/kkaiwwana/mustango) with >1min demo audio provided (I personaly think that sounds not bad).

hi @dorienh, would you like to take a look? And if you found it interseting, I'm happy to send a PR as a plug-and-play enhancement to the current verison of mustango.

**Comment by dorienh** [comment]
@kkaiwwana This is interesting, thanks for sharing. Do you happen to have any output examples. Will look into your paper.

**Comment by kkaiwwana** [comment]
@dorienh yes, please check this forked [repo](https://github.com/kkaiwwana/mustango), I provided many generated [examples](https://github.com/kkaiwwana/mustango/tree/main/assets/demo_audio).
